### PR TITLE
Add summary page, tests, check style, amend error page

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**

### CODE REVIEW

File: src/main/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientContactDetailsValidator.java [hunk 1]
```diff
@@ -79,6 +79,7 @@ private void validateTelephoneField(
    */
   private void validateEmailField(ClientDetails clientDetails, Errors errors) {
     if (!StringUtils.hasText(clientDetails.getEmailAddress())
+        && clientDetails.getCorrespondenceMethod() != null
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientContactDetailsValidator.java:82]: Pedantic I know, but if you reverse it then you don't need the null check.
CORRESPONDENCE_EMAIL.equalsIgnoreCase(clientDetails.getCorrespondenceMethod())

File: src/main/java/uk/gov/laa/ccms/caab/client/EbsApiClient.java [hunk 2]
```diff
@@ -88,6 +88,16 @@ public Mono<CommonLookupDetail> getCommonValues(String type) {
     return this.getCommonValues(type, null, null);
   }
 
+  /**
+   * Retrieves common lookup values based on the supplied type and code.
+   *
+   * @param type The type of the common lookup values. Can be null.
+   * @return A Mono containing the CommonLookupDetail or an error handler if an error occurs.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/client/EbsApiClient.java:95]: Need a @param for code

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/client/ClientSummaryController.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      Model model) {
+    log.info("GET /application/client/details/summary");
+
+    populateSummaryListLookups(clientDetails, model);
+
+    return "application/client/client-summary-details";
+  }
+
+  /**
+   * Handles the GET request for the client summary page.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/controller/application/client/ClientSummaryController.java:67]: Update for POST info

File: src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java [hunk 4]
```diff
@@ -87,6 +114,19 @@ public Mono<CommonLookupDetail> getMaritalStatuses() {
     return ebsApiClient.getCommonValues(COMMON_VALUE_MARITAL_STATUS);
   }
 
+  /**
+   * Get a Marital Status Common Value.
+   *
+   * @return CommonLookupDetail containing the common lookup values.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java:120]: Should have an @param for code

File: src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java [hunk 5]
```diff
@@ -109,6 +149,26 @@ public Mono<CommonLookupDetail> getCountries() {
         });
   }
 
+  /**
+   * Get a Country Common Value.
+   *
+   * @return CommonLookupValueDetail containing the common lookup value.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java:155]: Should have an @param for code

File: src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java [hunk 6]
```diff
@@ -118,6 +178,19 @@ public Mono<CommonLookupDetail> getCorrespondenceMethods() {
     return ebsApiClient.getCommonValues(COMMON_VALUE_CORRESPONDENCE_METHOD);
   }
 
+  /**
+   * Get a Correspondence Method Common Values.
+   *
+   * @return CommonLookupValueDetail containing the common lookup value.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java:184]: Should have an @param for code

File: src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java [hunk 7]
```diff
@@ -127,6 +200,19 @@ public Mono<CommonLookupDetail> getCorrespondenceLanguages() {
     return ebsApiClient.getCommonValues(COMMON_VALUE_CORRESPONDENCE_LANGUAGE);
   }
 
+  /**
+   * Get a Correspondence Language Common Values.
+   *
+   * @return CommonLookupValueDetail containing the common lookup value.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java:206]: Should have an @param for code

File: src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java [hunk 8]
```diff
@@ -136,6 +222,19 @@ public Mono<CommonLookupDetail> getEthnicOrigins() {
     return ebsApiClient.getCommonValues(COMMON_VALUE_ETHNIC_ORIGIN);
   }
 
+  /**
+   * Get an Ethnic Origin Common Values.
+   *
+   * @return CommonLookupValueDetail containing the common lookup value.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java:228]: Should have an @param for code

File: src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java [hunk 9]
```diff
@@ -145,5 +244,20 @@ public Mono<CommonLookupDetail> getDisabilities() {
     return ebsApiClient.getCommonValues(COMMON_VALUE_DISABILITY);
   }
 
+  /**
+   * Get a Disability Common Values.
+   *
+   * @return CommonLookupValueDetail containing the common lookup value.
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/service/CommonLookupService.java:250]: Should have an @param for code

File: src/main/resources/templates/application/client/client-summary-details.html [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+        <a href="/application/client/search" class="govuk-back-link">Cancel and return to client search</a>
+    </div>
+    <div id="main-content">
+        <form action="#" th:action="@{/application/client/details/summary}" th:object="${clientDetails}" method="post">
+            <span class="govuk-caption-l">About your client</span>
+            <h1 class="govuk-heading-l">
+                Summary of client information
+            </h1>
+            <p class="govuk-body-l">
+                You are about to create a new client record, after which you may go back and edit
```
Comment (porritta) [inline src/main/resources/templates/application/client/client-summary-details.html:18]: Is this the original text?!
"after which you may go back and edit any details that change" doesn't make sense to me.
And I presume it should be "request deletion of the record FROM the LAA"?

Comment (PhilDigitalJustice) [reply src/main/resources/templates/application/client/client-summary-details.html:18]: The from needs changing but the rest of the text is the same

You are about to create a new client record, after which you may go back and edit any details that change, or request deletion of the record from the LAA. Please check the summary details are correct before registration.

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**